packages/components/src/Combobox/Combobox.test.tsx

```diff
@@ -247,15 +251,185 @@ describe("Combobox selected value", () => {
+
+    expect(getByTestId("ATL-Combobox-Content")).not.toHaveClass("hidden");
   });
 
+  it("should allow for multiple selections to be made", () => {
+    const { getByText } = render(<MockMultiSelectOnSelectCombobox />);
+
+    const button = getByText("Click Me");
+    fireEvent.click(button);
+
+    const option = getByText("Bilbo Baggins");
+    const option2 = getByText("Frodo Baggins");
+
+    fireEvent.click(option);
+    fireEvent.click(option2);
+
+    expect(option).toHaveClass("selectedOption");
+    expect(option2).toHaveClass("selectedOption");
+  });
+  it("should call the onSelect callback upon making selection(s)", () => {
+    const onSelect = jest.fn();
+    const { getByText } = render(
+      <MockMultiSelectOnSelectCombobox onSelectOverride={onSelect} />,
+    );
+
+    const button = getByText("Click Me");
+    const option = getByText("Bilbo Baggins");
+
+    fireEvent.click(button);
+    fireEvent.click(option);
+
+    expect(onSelect).toHaveBeenCalledWith([
+      {
+        id: "1",
+        label: "Bilbo Baggins",
+      },
+    ]);
+  });
+  it("should not clear search after making a selection", () => {
+    const { getByText, getByPlaceholderText } = render(
+      <MockMultiSelectOnSelectCombobox />,
+    );
+
+    const button = getByText("Click Me");
+    const option = getByText("Bilbo Baggins");
+    const searchInput = getByPlaceholderText("Search");
+
+    fireEvent.click(button);
+    fireEvent.change(searchInput, { target: { value: "Bilbo" } });
+    fireEvent.click(option);
+
+    expect(searchInput).toHaveValue("Bilbo");
+  });
+
+  describe("onClose callback", () => {
+    it("should call onClose with selections when the content is closed", async () => {
+      const onClose = jest.fn();
+      const { getByText } = render(
+        <MockMultiSelectOnCloseCombobox onCloseOverride={onClose} />,
+      );
+
+      const button = getByText("Click Me");
+      const option = getByText("Bilbo Baggins");
+      const spoder = getByText("Shelob the Spoder");
+
+      fireEvent.click(button);
+      fireEvent.click(option);
+      fireEvent.click(spoder);
+      fireEvent.keyDown(button, { key: "Escape" });
+
+      expect(onClose).toHaveBeenCalledWith([
+        { id: "1", label: "Bilbo Baggins" },
+        { id: "3", label: "Shelob the Spoder" },
+      ]);
+    });
+    it("should not update consumer as selections are made", () => {
+      const { getByText, queryByText } = render(
+        <MockMultiSelectOnCloseCombobox />,
+      );
+
+      const button = getByText("Click Me");
+      const option = getByText("Bilbo Baggins");
+      const spoder = getByText("Shelob the Spoder");
+
+      fireEvent.click(button);
+      fireEvent.click(option);
+      fireEvent.click(spoder);
+
+      expect(queryByText("Choice: Bilbo Baggins")).not.toBeInTheDocument();
+      expect(queryByText("Choice: Shelob the Spoder")).not.toBeInTheDocument();
+    });
+  });
+});
+
+function MockMultiSelectOnCloseCombobox(props: {
```

Comment:
this got a bit acceptance-y

part of me wants to find ways to make them more unit or integration style tests, but also I feel like due to the nature of how the components work now, where the state is external and our selections depend on that state changing - we kinda have to do something like this